### DISCUSSION TITLE
How to make repo appear in Github after cloning VS 2022

### DISCUSSION THREAD

**Discussion by NapalmPig** [community/community#62445] [New to GitHub] [answered]
Hi all

Completely new to Github. I'm trying clone a repo to use as a starting point for various small projects.

I managed to clone it using the http link into VS, but I don't know how to get it to appear in my Github. I thought maybe to create a repo in Github and see if I could connect them, but that sounded like the sort of thing that would go wrong.

Any help appreciated

**Comment by h7x4** [comment, marked as answer]
Hello @NapalmPig. Welcome to GitHub!

What you are describing, sounds like a [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo). You are correct with the idea that you could do it manually, but that it's easier to use the web functionality. In this case, you're looking for this button:

![image](https://github.com/community/community/assets/14929991/bfba0a5d-f938-48ac-a9ff-77f878f702f7)

When you click this on someone elses user, it will create a copy in your own list of repositories. You are free to commit, push and pull as much as you'd like to this repo. (Note that this also means you have to clone the repository again, unless you want to get your hands dirty with git remotes)

Also note that depending on the license of the project you are forking, legal restrictions may apply to what you can do with the project. But most open source repositories have quite loose restrictions when it comes to forking and modifying the code, the restrictions are usually mostly related to publishing, copyright and commerce. Theres's a quick overview of some of the more common licenses here: https://choosealicense.com/licenses/

As for forking something manually, if you are interested in that, [here's some documentation on how git remotes work](https://www.atlassian.com/git/tutorials/syncing).

**Reply by NapalmPig** [reply]
Thanks. I made a fork as you said, to test it I made a simple change and pressed commit and then push, but it seems to have gone to the original owners version, not mine. No harm will come of it but I feel a bit silly. How do I work on my own separate instance? I feel I'm missing something crucial with all this.

**Reply by h7x4** [reply]
No worries. AFAIK, you shouldn't have access to the original owner version unless they have explicitly invited you as a maintainer of the project, so no harm done!

The easiest way of doing this is to delete your existing clone, and reclone it, only this time you use your own fork.

To verify that you have cloned the correct repository, you can run `git remote get-url origin` in a terminal inside your cloned repository, and check that the url it prints out points at your repository

**Reply by NapalmPig** [reply]
Tried typing that into the terminal and it said git wasn't recognised.

The commits that appeared on the original owners repo seem to have gone now. Im a bit nervous of doing anything more as I don't want to be spamming them with nonsense.

**Reply by h7x4** [reply]
> Tried typing that into the terminal and it said git wasn't recognised.

It sounds like you don't have git installed systemwide. While it's strictly not necessary, as Visual Studio includes git support natively, it might be useful to have for some kinds of tasks. Have a look here: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git

> The commits that appeared on the original owners repo seem to have gone now. Im a bit nervous of doing anything more as I don't want to be spamming them with nonsense.

I really don't think you should worry about this. You do not have authorization to make any edits to the source code of others, so it is completely safe to play around. Unless you make an issue or a pull request (and this is hard to do by accident), they will not even be notified that you are playing around.

It seems like you have successfully created a test branch on your own fork: https://github.com/NapalmPig/chili_framework/tree/test_branch1

Congrats! :tada:
